rust/rsmgp-sys/src/value/mod.rs

```diff
@@ -0,0 +1,326 @@
+            return Err(MgpError::UnableToAllocatePathValue);
+        }
+        let mgp_value = ffi::mgp_value_make_path(mgp_copy);
+        if mgp_value.is_null() {
+            ffi::mgp_path_destroy(mgp_copy);
+            return Err(MgpError::UnableToAllocatePathValue);
+        }
+        Ok(MgpValue { ptr: mgp_value })
+    }
+}
+
+#[derive(Debug)]
+pub enum Value {
+    Null,
+    Bool(bool),
+    Int(i64),
+    Float(f64),
+    String(CString),
+    Vertex(Vertex),
+    Edge(Edge),
+    Path(Path),
+}
+
+impl Value {
+    // TODO(gitbuda): Remove to_result_mgp_value dead code.
+    #[allow(dead_code)]
+    fn to_result_mgp_value(&self, context: &Memgraph) -> MgpResult<MgpValue> {
+        match self {
+            Value::Null => make_null_value(context),
+            Value::Bool(x) => make_bool_value(*x, context),
+            Value::Int(x) => make_int_value(*x, context),
+            Value::String(x) => make_string_value(&*x.as_c_str(), context),
+            Value::Float(x) => make_double_value(*x, context),
+            Value::Vertex(x) => make_vertex_value(&x, context),
+            Value::Edge(x) => make_edge_value(&x, context),
+            Value::Path(x) => make_path_value(&x, context),
+        }
+    }
+}
+
+/// # Safety
+/// TODO(gitbuda): Write section about safety.
+pub unsafe fn make_vertex_copy(
+    mgp_vertex: *const mgp_vertex,
+    context: &Memgraph,
+) -> MgpResult<Vertex> {
+    assert!(
+        !mgp_vertex.is_null(),
+        "Unable to make vertex copy because vertex is null."
+    );
+    let mgp_vertex_copy = ffi::mgp_vertex_copy(mgp_vertex, context.memory());
+    if mgp_vertex_copy.is_null() {
+        return Err(MgpError::UnableToAllocateVertexValue);
+    }
+    Ok(Vertex {
+        ptr: mgp_vertex_copy,
+        context: context.clone(),
+    })
+}
+
+/// # Safety
+/// TODO(gitbuda): Write section about safety.
+pub unsafe fn make_edge_copy(mgp_edge: *const mgp_edge, context: &Memgraph) -> MgpResult<Edge> {
+    assert!(
+        !mgp_edge.is_null(),
+        "Unable to make edge copy because edge is null."
+    );
+    let mgp_edge_copy = ffi::mgp_edge_copy(mgp_edge, context.memory());
+    if mgp_edge_copy.is_null() {
+        return Err(MgpError::UnableToAllocateEdgeValue);
+    }
+    Ok(Edge {
+        ptr: mgp_edge_copy,
+        context: context.clone(),
+    })
+}
+
+/// # Safety
+/// TODO(gitbuda): Write section about safety.
+pub unsafe fn make_path_copy(mgp_path: *const mgp_path, context: &Memgraph) -> MgpResult<Path> {
+    assert!(
+        !mgp_path.is_null(),
+        "Unable to make path copy because path is null."
+    );
+    let mgp_path_copy = ffi::mgp_path_copy(mgp_path, context.memory());
+    if mgp_path_copy.is_null() {
+        return Err(MgpError::UnableToAllocatePathValue);
+    }
+    Ok(Path {
+        ptr: mgp_path_copy,
+        context: context.clone(),
+    })
+}
+
+/// # Safety
+/// TODO(gitbuda): Write section about safety.
+pub unsafe fn mgp_raw_value_to_value(
```

Comment:
Result of this should be MgpValue that stores value which can be interpreted different (value.int_value(), value.double_value() ...)

Reply:
Yea, this is I think the same discussion about `MgpValue` and `Value`, once the decision about values will be in place, this will be updated accordingly :smile: